pylint/checkers/classes/class_checker.py

```diff
@@ -1495,11 +1495,17 @@ def _check_in_slots(self, node):
                     # Properties circumvent the slots mechanism,
                     # so we should not emit a warning for them.
                     return
-                if node.attrname in klass.locals and _has_data_descriptor(
-                    klass, node.attrname
-                ):
-                    # Descriptors circumvent the slots mechanism as well.
-                    return
+                if node.attrname in klass.locals:
+                    for local_name in klass.locals.get(node.attrname, []):
```

Comment:
```suggestion
                if node.attrname in klass.locals:
                    for local_name in klass.locals.get(node.attrname):
```
```suggestion
                for local_name in klass.locals.get(node.attrname, []):
```
We know `node.attrname` is in `klass.locals` at this point